### ISSUE TITLE
Performance Preview - WaterFall Chart

### ISSUE THREAD

**Issue by MarkForesta** [github/vscode-github-actions#407]
**Is your feature request related to a problem? Please describe.**

I think CI/CD systems can get extremely complex. I don't think it's that crazy to believe that some workflows will run 3-5 machines in parallel and branch up to around 20 jobs. I work at a company that has very rigorous compliance standards and I have a goal to move all of our compliance steps away from the primary thread our users will do work on. Ultimately trying to speed up pipelines and better devex. I think it would be amazing if we could visualize this better.

**Describe the solution you'd like**

This requirement sent me down a rabbit hole thinking: What if when I changed a workflow file locally and it showed me a preview of how more performant my workflow run would be? Something like Chrome Dev Tools network tab?
<img width="843" alt="image" src="https://github.com/user-attachments/assets/472bfed5-2da2-4d83-8311-747207e1b08c"> 

Then I could even move a task (Single Github action step) to another spot in the workflow file and that chart would change automatically. Maybe change the title to clearly show this is now a preview if your change worked.

I think if we really nailed this implementation I think it could save complex workflow writers a lot of time.

So I would add a new command to show this chart in a preview (New tab to the right in VSCode).

**Additional context**
First and foremost I know this is kind of a crazy ask. So I am totally ok with you saying: Nah this is crazy go do this in a separate extension. I just feel since you are already authenticated and making the API request it's the right place for it.

Secondly this might be something better suited for the actual web ui. Even if it didn't auto update based on local changes to a workflow file, this view of a workflow might still be useful. 

I know this is a bit half baked but I am really trying to gauge if other users would find this interesting.

**If this issue was accepted as a feature request to the product and I was working on it I would split this into 2 parts**

1. Find a way to take existing workflow runs and create a performance preview in JSON or some other data format based off a potential workflow file change.
2. Find a way to visualize existing workflow executions as a waterfall chart or something that can consume this data. 

PS: If people liked this idea I would definitely be interested in contributing the change
